pkg/syncer/spec/spec_controller.go

```diff
@@ -98,6 +98,7 @@ func NewSpecSyncer(gvrs []schema.GroupVersionResource, upstreamClusterName logic
 		})
 		klog.InfoS("Set up informer", "clusterName", upstreamClusterName, "pcluster", workloadClusterName, "gvr", gvr.String())
 	}
+	upstreamInformers.ForResource(schema.GroupVersionResource{Version: "v1", Resource: "namespaces"}).Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{})
```

Comment:
the last call afaik is not necessary.

Reply:
well, when not there the namespaces are not present in the informer cache.

Reply:
Alternative, much more in line with other code: have a namespaceLister in the controller object, and here you would call Lister() instead of Informer().

Reply:
1046 uses the lister but is blocked on other things

Reply:
I can change for the lister option, which would allow merging this fix without waiting for the rest of the other PR ?

Reply:
Done.